core/src/main/java/org/apache/iceberg/data/avro/DataWriter.java

```diff
@@ -50,13 +50,17 @@ public void setSchema(Schema schema) {
     this.writer = (ValueWriter<T>) AvroSchemaVisitor.visit(schema, new WriteBuilder());
   }
 
+  protected void setWriter(ValueWriter<T> writer) {
+    this.writer = writer;
```

Comment:
Should `setWriter` be used in `setSchema` above for this class as well?

Reply:
The `setWriter` can be removed now in the new patch because we don't need to expose it in new way.